csharp/src/Ice/UdpTransceiver.cs

```diff
@@ -586,14 +585,12 @@ public ConnectionInfo GetInfo()
 
         public void CheckSendSize(int size)
         {
-            //
-            // The maximum packetSize is either the maximum allowable UDP packet size, or
-            // the UDP send buffer size (which ever is smaller).
-            //
+            // The maximum packetSize is either the maximum allowable UDP packet size, or the UDP send buffer size
+            // (which ever is smaller).
             int packetSize = Math.Min(MaxPacketSize, _sndSize - UdpOverhead);
             if (packetSize < size)
             {
-                throw new DatagramLimitException();
+                throw new DatagramLimitException($"cannot send more than {packetSize} bytes with UDP");
             }
         }
```

Comment:
I think we should get rid of this check and the DatagramLimitException. We should recommend the user to get the transport send buffer size through the connection information and he should make sure he doesn't sent datagrams larger than the packet size computed by the function (we could eventually improve the connection information to provide this packet size...). If you're using UDP you'd better be aware of these limits and deal with them upfront.

Reply:
I agree, for a future PR.